.github/workflows/codeigniter.yml

```diff
@@ -10,11 +10,51 @@ on:
   pull_request:
     branches: [ master, devel ]
 jobs:
+  required_php_versions:
+    name: Get PHP versions to test
+    runs-on: ubuntu-latest
+    outputs:
+      php_versions: ${{ steps.php_ver_step.outputs.PHP_VERSIONS }}
+      php_versions_matrix: ${{ steps.php_ver_step.outputs.PHP_VERSIONS_matrix }}
+    steps:
+      - name: Install required packages
+        run: |
+          if ! command -v jq; then sudo apt-get update && sudo apt-get install -y jq; fi
+      - name: Build PHP_VERSIONS array & PHP_VERSIONS_matrix
+        id: php_ver_step
+        run: |
+          set -x
+
+          # Set array that will store the PHP versions for which we create a package.
+          PHP_VERSIONS=()
+
+          # Get all released php versions above 5.6 (in the format X.Y)
+          for upstream_ver in $(curl https://www.php.net/releases/?json | jq -r '.[].version' | cut -f -2 -d .); do
+            major=$(cut -f 1 -d . <<< "$upstream_ver")
+            for minor in {0..20}; do
+              if dpkg --compare-versions ${major}.$minor le $upstream_ver &&  dpkg --compare-versions ${major}.$minor ge 5.6; then
+                PHP_VERSIONS+=("${major}.$minor")
+              fi
+            done
+          done
```

Comment:
Is there any other simpler way to identify the versions? Maybe we could have fixed values that we support? My only concern is that is is a version released which is not supported, suddenly the CI would stop working, and throwing failure, even when other versions are supported.

Reply:
Yes, this seems better. Stable version shouldn't fail and would get rightly reported

Reply:
Correct

Reply:
If we have only stable versions, then all good